apps/workers/bigquery/write.go

```diff
@@ -0,0 +1,184 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+
+	"github.com/pcelvng/task-tools/file"
+	"github.com/pcelvng/task-tools/file/stat"
+)
+
+func writeToFile(ctx context.Context, j *bigquery.Job, w file.Writer, format string) (sts stat.Stats, err error) {
+	rows, err := j.Read(ctx)
+	if err != nil {
+		return sts, fmt.Errorf("row: %w", err)
+	}
+
+	loader := &bqValueLoader{}
+
+	// Set header for CSV format
+	if format == "csv" {
+		schema := rows.Schema
+		header := make([]string, len(schema))
+		for i, field := range schema {
+			header[i] = field.Name
+		}
+		loader.Header = header
+
+		// Write header row
+		err = w.WriteLine([]byte(strings.Join(header, ",")))
+		if err != nil {
+			return sts, fmt.Errorf("write header: %w", err)
+		}
+	}
+
+	for err = rows.Next(loader); err == nil; err = rows.Next(loader) {
```

Comment:
The loop terminates when rows.Next() returns an error, but the final error (other than io.EOF) isn't logged or returned explicitly.
```go
if err != nil && err != io.EOF {
    return sts, fmt.Errorf("row iteration error: %w", err)
}
```